transaction/core/src/signer/local_signer.rs

```diff
@@ -0,0 +1,66 @@
+use super::{Error, RingSigner};
+use crate::{
+    onetime_keys::recover_onetime_private_key,
+    ring_signature::{generators, CryptoRngCore, RingMLSAG, Scalar, SignableInputRing},
+};
+use mc_account_keys::AccountKey;
+use mc_crypto_keys::RistrettoPublic;
+
+/// An implementation of RingSigner that holds private keys
+#[derive(Clone, Debug)]
+pub struct LocalRingSigner {
+    key: AccountKey,
```

Comment:
If we don't want `RingSigner` to take `&mut RngCore + CryptoRng` as an argument, then the local ring signer must contain an RNG or a reference to an RNG.

That could work, but it will mean that `RingSigner::sign` must take `&mut self` rather than `&self`, so it won't be thread-safe anymore. It seems desirable for it to be thread-safe.

It could own an RNG of it's own behind a mutex, but then this crate will not be `no_std` compatible.

Therefore we decided that the function should take a reference to an RNG as a parameter instead.

Reply:
The downside may be just, having to futz with trait objects?

Full disclosure, I think that's the way to go

Reply:
Me too